src/session.c

```diff
@@ -126,6 +125,7 @@ static void trim_padded_field(CK_UTF8CHAR *field, ssize_t n)
 }
 
 #define trim(x) trim_padded_field(x, sizeof(x))
+const char *slot_desc_fmt = "PKCS#11 Token (Slot %lu - %s)";
```

Comment:
```suggestion
static const char slot_desc_fmt[] = "PKCS#11 Token (Slot %lu - %s)";
```

Otherwise `sizeof(slot_desc_fmt)` below will always return a pointer size.

Reply:
nice catch
amazing how broken things seemingly work :)